### PR TITLE
Added log file dedicated to maven executions (per GAV)

### PR DESCRIPTION
**Pull request by genadi-hp**
Sign-off: @genadi-hp, Genadi Rabinovich, genadi@hpe.com

### CODE REVIEW

File: dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/DependencyServiceImpl.java [hunk 1]
```diff
@@ -65,7 +74,40 @@ private void initMaven() throws ClassNotFoundException, NoSuchMethodException, M
...
+            MAVEN_EXECUTE_METHOD = Class.forName(MAVEN_LAUNCHER_CLASS_NAME, true, mavenClassLoader).
+                    getMethod(MAVEN_LANUCHER_METHOD_NAME, String[].class);
+            initMavenLogs();
+        }
+    }
+
+    private void initMavenLogs() {
+        String logFolderPath = calculateLogFolderPath();
+        File logFolderFile = new File(logFolderPath);
+        logFolderFile.mkdirs();
```
Comment (Bonczidai) [inline dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/DependencyServiceImpl.java]: do we need both `mkdir`s?

File: dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/DependencyServiceImpl.java [hunk 2]
```diff
@@ -65,7 +74,40 @@ private void initMaven() throws ClassNotFoundException, NoSuchMethodException, M
...
+    private String calculateLogFolderPath() {
+        Enumeration e = Logger.getRootLogger().getAllAppenders();
+        while ( e.hasMoreElements() ){
+            Appender app = (Appender)e.nextElement();
+            if ( app instanceof FileAppender){
+                String logFile = ((FileAppender) app).getFile();
+                return new File(logFile).getParentFile().getAbsolutePath();
+            }
         }
+        return new File(System.getProperty("app.home"), "logs").getAbsolutePath();
```
Comment (Bonczidai) [inline dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/DependencyServiceImpl.java]: IMHO it's worth extracting these strings as constants: app.home, logs, maven etc..

Comment (Bonczidai) [inline dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/DependencyServiceImpl.java]: they already used in several places in the codebase

**Comment by Bonczidai** [comment]
:+1: looks good - please handle comments

File: dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/DependencyServiceImpl.java (hunk 1 above)
Comment (genadi-hp) [inline dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/DependencyServiceImpl.java]: done

File: dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/DependencyServiceImpl.java (hunk 2 above)
Comment (genadi-hp) [inline dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/DependencyServiceImpl.java]: I did not find app.home in score (may be they are present in cloudslang?)
Meanwhile i defined local constants for future reuse

**Comment by Bonczidai** [comment]
:ok_hand:

**Pull request merged**